### PR TITLE
Add JWQL Python 3.9 Environment

### PR DESCRIPTION
**Pull request by mfixstsci**
- Created 3.9 environment
- Removed support for python 3.7
- Updated README docs

### CODE REVIEW

**Comment by mfixstsci** [comment]
@bhilbert4 I went ahead and updated the python 3.9 environment file to match 3.8. Since April is around the corner I figured we could just wait to see what pyup has to say. This is ready for review.

File: environment_python_3_9.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ dependencies:
   - psycopg2=2.8.6
   - pytest=6.2.4
   - pytest-cov=3.0.0
-  - python=3.7.10
+  - python=3.8.5
```
Comment (bhilbert4) [inline environment_python_3_9.yml:45]: This should be 3.9.11?

**Review by bhilbert4 — APPROVED** [review]
Looks good to me. I'm guessing the next dependency update will probably include the same updates from #896, plus whatever is new since then.

File: environment_python_3_9.yml (hunk 1 above)
Comment (mfixstsci) [reply environment_python_3_9.yml:45]: Looks like I missed this, I'll get on it and merge.

**Pull request merged**